server/gitlab.go

```diff
@@ -161,6 +165,11 @@ func hasSameEnvs(info *E2ETestTriggerInfo, glVars []*gitlab.PipelineVariable) (b
 				}
 				i++
 			}
+			// Triggering PR defaults to 6 created default pipeline environment variables.
+			// See triggerE2EGitLabPipeline in gitlab.go . Adding a custom env vars has 7.
+			if len(glVars) > 6 {
+				return false, nil
+			}
```

Comment:
Btw, I feel like the comparison inside `if info.EnvVars == nil {...}` could be omitted since it's likely caught by the next for-range loop. Could that be replaced by the following?
```go
if len(info.EnvVars) != len(glVars) {
    return false, nil
}
```